src/tablestore.rs

```diff
@@ -163,10 +170,32 @@ impl TableStore {
         for chunk in encoded_sst.unconsumed_blocks {
             data.put_slice(chunk.as_ref())
         }
-        self.object_store
-            .put(&path, PutPayload::from(data))
-            .await
-            .map_err(SlateDBError::ObjectStoreError)?;
+
+        match id {
+            SsTableId::Wal(wal_id) => {
+                let path = Path::from(format!("{:020}.sst", wal_id,));
+                match self
+                    .transactional_wal_store
+                    .put_if_not_exists(&path, Bytes::from(data))
+                    .await
+                {
+                    Ok(_) => (),
+                    Err(e) => match e {
+                        object_store::Error::AlreadyExists { path: _, source: _ } => {
+                            return Err(SlateDBError::Fenced)
+                        }
+                        _ => return Err(SlateDBError::ObjectStoreError(e)),
+                    },
+                }
+            }
+            SsTableId::Compacted(_) => {
```

Comment:
Curious about this. Wouldn't we want compactor writes to be fenced as well? They're ULIDs names, so we'd want to detect if they're ever going to be over written.

Reply:
Using transactional store for both now, and returning Fenced error for wal. Compactor writes I believe would need to be retried, name collision alone is not sufficient to say that compactor is fenced.